packages/langium/src/utils/promise-util.ts

```diff
@@ -84,22 +84,25 @@ export async function interruptAndCheck(token: CancellationToken): Promise<void>
 export class MutexLock {
 
     private previousAction = Promise.resolve();
+    private previousTokenSource = new CancellationTokenSource();
 
     /**
      * Performs a single async action, like initializing the workspace or processing document changes.
      * Only one action will be executed at a time.
      */
-    async lock(action: () => Promise<void>): Promise<void> {
+    lock(action: (token: CancellationToken) => Promise<void>): Promise<void> {
```

Comment:
Testing this would be great. Also for documenting how it is used...

Reply:
I've added some additional tests for this. The `lock` function is already tested, but the cancellation wasn't yet.